### ISSUE TITLE
SchedulerPredicates validates MaxPods and SchedulerPredicates validates resource limits failing consistently in the soak test

### ISSUE THREAD

**Issue by roberthbailey** [kubernetes/kubernetes#15334]
Similar to https://github.com/kubernetes/kubernetes/issues/15139 but rather than flaking, the tests are failing on every run. 

```
Kubernetes e2e suite.SchedulerPredicates validates MaxPods limit number of pods that are allowed to run.￼ (Failed 30 times in the last 30 runs. Stability: 0 %)￼
Kubernetes e2e suite.SchedulerPredicates validates resource limits of pods that are allowed to run.￼ (Failed 30 times in the last 30 runs. Stability: 0 %)￼
```

**Comment by gmarek** [comment]
I'm trying to reproduce the failure locally but for some reason I cannot.

**Comment by gmarek** [comment]
My guess is that it's kind of bad luck. We had a problem introduced by #15022 and fixed by #15144, but sadly Soak cluster was built between those two. I'll kick the build of soak cluster, so it'll catch up the fix.
